infrastructure/identity-aware-proxy/iap_validating_server.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example of verifying IAP signed headers in web requests."""
+
+import requests
+from BaseHTTPServer import BaseHTTPRequestHandler
+from BaseHTTPServer import HTTPServer
+from googleapiclient import discovery
+# validate_jwt github link:
+#    https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/iap/validate_jwt.py
+import validate_jwt
+
+
+class RequestHandler(BaseHTTPRequestHandler):
+  project_number = None
+  backend_service_id = None
+
+  def do_GET(self):
+    """Intercepts all GET requests and validates
+    the IAP JWT that is present in the header.
+    
+    """
+    print self.headers
+    identity = validate_jwt.validate_iap_jwt_from_compute_engine(
+      self.headers.get("X-Goog-IAP-JWT-Assertion"),
+      self.project_number,
+      self.backend_service_id)
+    if not identity[1]:
+      self.send_response(200)
```

Comment:
403s cause http health checker to fail and in addition, returning 200 allows me to display a message to the user about the current state of the web app. This will be more useful for demo purposes

Reply:
You can return whatever content you like along with a 403 response, just like a 200. You're right about the health checker falling over, though. SGTM.